Makefile

```diff
@@ -0,0 +1,153 @@
+PACKAGES=$(shell go list ./... | grep -v '/vendor/')
+PACKAGES_NOCLITEST=$(shell go list ./... | grep -v '/vendor/' | grep -v github.com/cosmos/ethermint/cmd/gaia/cli_test)
+COMMIT_HASH := $(shell git rev-parse --short HEAD)
+BUILD_FLAGS = -tags netgo -ldflags "-X github.com/cosmos/ethermint/version.GitCommit=${COMMIT_HASH}"
+
+DEP = github.com/golang/dep/cmd/dep
+GOLINT = github.com/tendermint/lint/golint
+GOMETALINTER = gopkg.in/alecthomas/gometalinter.v2
+UNCONVERT = github.com/mdempsky/unconvert
+INEFFASSIGN = github.com/gordonklaus/ineffassign
+MISSPELL = github.com/client9/misspell/cmd/misspell
+ERRCHECK = github.com/kisielk/errcheck
+UNPARAM = mvdan.cc/unparam
+GOCYCLO = github.com/alecthomas/gocyclo
+
+DEP_CHECK := $(shell command -v dep 2> /dev/null)
+GOLINT_CHECK := $(shell command -v golint 2> /dev/null)
+GOMETALINTER_CHECK := $(shell command -v gometalinter.v2 2> /dev/null)
+UNCONVERT_CHECK := $(shell command -v unconvert 2> /dev/null)
+INEFFASSIGN_CHECK := $(shell command -v ineffassign 2> /dev/null)
+MISSPELL_CHECK := $(shell command -v misspell 2> /dev/null)
+ERRCHECK_CHECK := $(shell command -v errcheck 2> /dev/null)
+UNPARAM_CHECK := $(shell command -v unparam 2> /dev/null)
+GOCYCLO_CHECK := $(shell command -v gocyclo 2> /dev/null)
+
+all: get-tools get-vendor-deps install
+
+########################################
+### CI
+
+ci: get-tools get-vendor-deps install
+
+########################################
+### Build
+
+# This can be unified later, here for easy demos
+build:
+ifeq ($(OS),Windows_NT)
+	go build $(BUILD_FLAGS) -o build/ethermint.exe ./*.go
+else
+	go build $(BUILD_FLAGS) -o build/ethermint ./*.go
+endif
+
+install:
+	go install $(BUILD_FLAGS) ./*.go
+
```

Comment:
Can we also add a `clean` target, that say, removes `./build` and `./vendor`?

Reply:
Yup!